pallene/to_ir.lua

```diff
@@ -21,7 +21,7 @@ typedecl.declare(to_ir, "to_ir", "LHS", {
 local ToIR = util.Class()
 
 function to_ir.convert(prog_ast)
-    local module = ToIR.new():convert_toplevel(prog_ast)
+    local module = ToIR.new():convert_toplevel(prog_ast.tls)
```

Comment:
Like the checker.convert_program, this should also have an assertion that the `prog_ast._tag` is what you expect it to be.